src/index.js

```diff
@@ -25,17 +26,35 @@ class ProfilerPlugin {
       'IOPIPE_ENABLE_PROFILER',
       this.config.enabled
     );
-    this.heapsnapshotEnabled = enabled(
+    this.heapEnabled = enabled(
       'IOPIPE_ENABLE_HEAPSNAPSHOT',
       this.config.heapSnapshot
     );
-    this.enabled = this.profilerEnabled || this.heapsnapshotEnabled;
+    this.enabled = this.profilerEnabled || this.heapEnabled;
     this.uploads = [];
 
+    // pre-invoke hooks cannot be async, so create our own promise to wait on
+    // before the post-invoke method runs
+    this.pluginReadyPromise = Promise.resolve();
+
     this.hooks = {
       'pre:invoke': this.preInvoke.bind(this),
-      'post:invoke': this.postInvoke.bind(this)
+      'post:invoke': this.postInvoke.bind(this),
+      'post:report': this.postReport.bind(this)
+    };
+    this.session = new inspector.Session();
+    // promisify session.post
```

Comment:
Any recommendations on where I can read up about why this needs to be promisified?

Reply:
It doesn't _need_ to be - but then we'd have callback hell everywhere, see:

Reply:
Ahh, ok, yeah, that's helpful context! Thank you!